### PR TITLE
Consolidate OpenSSL build into a single CMakeLists file

### PR DESCRIPTION
**Pull request by ThadHouse**
Currently has a regression where CMake will double build OpenSSL. Working on debugging this.

### CODE REVIEW

**Comment by thhous-msft** [comment]
/azp run

This is coming from my mac, which can't use my msft account. Anything mac related will be personal account.

File: submodules/CMakeLists.txt [hunk 1]
```diff
@@ -5,146 +5,270 @@ cmake_minimum_required(VERSION 3.16)
...
+            # The following line is needed for the 3.0 branch.
+            #
+            # no-uplink no-cmp no-acvp_tests no-fips no-padlockeng no-siv
+
+            no-zlib no-egd no-idea no-rc5 no-rc4 no-afalgeng
+            no-comp no-cms no-ct no-srp no-srtp no-ts no-gost no-dso no-ec2m
+            no-tls1 no-tls1_1 no-tls1_2 no-dtls no-dtls1 no-dtls1_2 no-ssl
+            no-ssl3-method no-tls1-method no-tls1_1-method no-tls1_2-method no-dtls1-method no-dtls1_2-method
+            no-siphash no-whirlpool no-aria no-bf no-blake2 no-sm2 no-sm3 no-sm4 no-camellia no-cast no-md4 no-mdc2 no-ocb no-rc2 no-rmd160 no-scrypt
+            no-weak-ssl-ciphers no-shared no-tests)
```
Comment (nibanks) [inline submodules/CMakeLists.txt:33]: nit: tab left some?

File: submodules/CMakeLists.txt [hunk 2]
```diff
@@ -5,146 +5,270 @@ cmake_minimum_required(VERSION 3.16)
...
+    else()
+        target_link_libraries(
+            OpenSSLQuic
+            INTERFACE
+            ${LIBCRYPTO_PATH}
+        )
+    endif()
+
+    add_library(OpenSSLQuic::OpenSSLQuic ALIAS OpenSSLQuic)
+endif()
```
Comment (nibanks) [inline submodules/CMakeLists.txt:274]: nit: add newline

**Review by nibanks — APPROVED** [review]

**Pull request merged**